draft-irtf-cfrg-hpke.md

```diff
@@ -644,13 +679,60 @@ octet strings for public keys.
 
 # Security Considerations
 
-[[ TODO ]]
+The general security properties of HPKE are described in
+{{security-properties}}.  In this section, we consider a security issue that may
+arise in practice and an advanced use case.
+
+## Metadata Protection
+
+The authenticated modes of HPKE (PSK, Auth, AuthPSK) require that the receiver
+know what key material to use for the initiator.  This can be signaled in
+applications by sending the PSK ID (`pskID` above) and/or the initiator's public
+key (`pkI`).  However, these values themselves might be considered sensitive,
+since in a given application context, they might identify the initiator.
+
+An application that wishes to protect these metadata values without requiring
+further provisioning of keys can use an additional instance of HPKE, using the
+unauthenticated base mode.  Where the application might have sent `(pskID, pkI,
+enc, ciphertext)` before, it would now send (enc2, ciphertext2, enc, ciphertext),
+where `(enc2, ciphertext2)` represent the encryption of the `pskID` and `pkI`
+values.
+
+The cost of this approach is an additional KEM operation each for the sender and
+the receiver.  A potential lower-cost approach (involving only symmetric
+operations) would be available if the nonce-protection schemes in {{BNT19}}
+could be extended to cover other metadata.  However, this construction would
+require further analysis.
+
+## Designated-Verifier Signature
```

Comment:
I wonder if this should be replaced with a more general "use cases" section in the document?

Reply:
Maybe, but I'm inclined to just do this for now.

Reply:
Yep, just food for thought in the future.